ui/app/controllers/vault/cluster/access/oidc/keys/key/details.js

```diff
@@ -0,0 +1,35 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+
+export default class OidcKeyDetailsController extends Controller {
+  @service router;
+  @service flashMessages;
+
+  @task
+  @waitFor
+  *rotateKey() {
```

Comment:
does it make sense for this to be in the controller? I was following the pattern of where we have `delete` but wasn't sure if that was correct here

Reply:
Yeah, I think so. The route/controller pattern seems to be model and hooks in the router file, and actions, tracked properties, methods that impact the rendering of the data in the controller. Unless it's easier to move this into a component, I would say the controller makes sense. But happy to hear anyone else's opinions on this. 

From the guides:
>Any attributes or actions that we want to share with components used on that Route could be defined on the Controller and passed down through the Route's template

Reply:
I think in general we prefer to keep actions in components, so it may make sense to make a Page component with all this logic

Reply:
I had noticed that was the pattern for the `keymgmt` secrets engine. It felt weird to add a whole new details component for only the `keys` route to handle the rotate action as all of the other `oidc-config` routes handle `delete` in their controller. But I can refactor!

Reply:
My understanding of the BP on controllers is that because they're singletons, one should use them only on tracking/modifying state derived from either queryParams or the model, otherwise you can have a bad time. In this case, the delete logic appears to be related to model state so it's probably ok, or at least a grey area/coin flip, which I guess is why we're discussing it :).